libraries/SPI/SPI.h

```diff
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 
 #define SPI_HAS_TRANSACTION
+#define SPI_SKIP_INITIAL_BUSY_CHECK //comment this line for backward compatibility if really needed
```

Comment:
I do think for backwards compatibility in the same way as you do.
But in fact I think that the initial check for `!busy` is not needed in any of the SPI functions as every write- and transfer-function already checks that its own SPI transfer completes before leaving its context by using
`while(SPI1CMD & SPIBUSY) {}`
at the end of the function.

Therefore it is not possible to change frequency during a transfer. 

I've only added the #define which can be commented as a service in case I've overlooked something. But currently I don't see any threat here, so maybe we can just delete the #define and the initial check completely.

Or is there an effect I currently don't know about that changing the frequency, data size or whatever activates SPI's busy flag? Then I would suggest to wait in the setFrequency or setDataBit functions but not in every transfer.